### ISSUE TITLE
Push/Pull file example

### ISSUE THREAD

**Issue by yuvalturg** [oras-project/oras-go#341]
Hi,

Is there an example for a simple push-pull of a single, large (can't load it's entire content to memory) file to a repository?

Thanks!

**Comment by Wwwsylvia** [comment]
Hi @yuvalturg, can you elaborate your scenario a little bit?
- Do you want to push/pull the file as a single blob?
- Do you want to have a manifest to reference this blob (file)?

Basically, how do you want to access this file after you pushing it to a repository?

**Comment by yuvalturg** [comment]
Yes a single layer is fine. Similar to what is described here https://oras.land/cli/1_pushing/

Oras push repo:tag /tmp/bigfile

**Comment by shizhMSFT** [comment]
@Wwwsylvia Could you write an example for @yuvalturg. Besides, we probably need `content.NewDescriptorFromReader()` to simplify the experience.

**Comment by yuvalturg** [comment]
Thanks, I used oras-py, and it's as simple as
OrasClient().push(target, filenames)

**Comment by Wwwsylvia** [comment]
@yuvalturg OK you will need a manifest to reference your file blob if you want to access it via a tag.
Can you check if this helps? https://github.com/Wwwsylvia/demo-push-file/blob/main/main/main.go
Please let us know if you find any bugs or any bad user experiences.

**Comment by Wwwsylvia** [comment]
> @Wwwsylvia Could you write an example for @yuvalturg? Besides, we probably need `content.NewDescriptorFromReader()` to simplify the experience.

Not really, I think a file store is needed anyway in this case 🤔

**Comment by yuvalturg** [comment]
@Wwwsylvia is there a way to get the local filename(s) from the filestore after pulling?

**Comment by Wwwsylvia** [comment]
> @Wwwsylvia is there a way to get the local filename(s) from the filestore after pulling?

@yuvalturg Regarding "get the local filename(s) from the filestore", my understanding is that you want to check the filenames associated with the pulled tag? Does the below code work for you?

```go
	desc, err := fs.Resolve(ctx, tag)
	if err != nil {
		panic(err)
	}
	blobs, err := content.Successors(ctx, fs, desc)
	if err != nil {
		panic(err)
	}
	for _, b := range blobs {
		if name := b.Annotations[v1.AnnotationTitle]; name != "" {
			fmt.Println("filename: ", name)
		}
	}
```

**Comment by yuvalturg** [comment]
> > @Wwwsylvia is there a way to get the local filename(s) from the filestore after pulling?
> 
> @yuvalturg Regarding "get the local filename(s) from the filestore", my understanding is that you want to check the filenames associated with the pulled tag? Does the below code work for you?
> 
> ```go
> 	desc, err := fs.Resolve(ctx, tag)
> 	if err != nil {
> 		panic(err)
> 	}
> 	blobs, err := content.Successors(ctx, fs, desc)
> 	if err != nil {
> 		panic(err)
> 	}
> 	for _, b := range blobs {
> 		if name := b.Annotations[v1.AnnotationTitle]; name != "" {
> 			fmt.Println("filename: ", name)
> 		}
> 	}
> ```

This works perfectly, thanks!

**Comment by Wwwsylvia** [comment]
Hello @yuvalturg, do you have any feedbacks on the user experiences of `oras-go`? 
For instance, do you think `file.PackFiles()` is useful? We are revisiting this function and are considering whether to keep/refactor/deprecate in the future versions.

**Comment by alishah730** [comment]
I can simply push any file using oras cli like below
`oras push file.rpm demo.goharbor.io/oras/test:latest`

I can't simply find any example using oras-go.

**Comment by Wwwsylvia** [comment]
We may want to add a few more examples into https://pkg.go.dev/oras.land/oras-go/v2@v2.0.0/content/file#pkg-examples.

**Comment by Wwwsylvia** [comment]
The push/pull file example is now available at https://oras.land/client_libraries/0_go/#quick-start.

**Comment by Wwwsylvia** [comment]
This is also tracked by #434

**Comment by shizhMSFT** [comment]
Closed by #434
